rest/route/pod_agent.go

```diff
@@ -443,44 +434,57 @@ func (h *podAgentEndTask) Run(ctx context.Context) gimlet.Responder {
 		})
 	}
 
-	// mark task as finished
 	deactivatePrevious := utility.FromBoolPtr(projectRef.DeactivatePrevious)
 	err = model.MarkEnd(t, evergreen.APIServerTaskActivator, finishTime, &h.details, deactivatePrevious)
 	if err != nil {
 		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "calling mark finish on task %v", t.Id))
 	}
 
+	// Clear the running task on the pod now that the task has finished.
+	if err = p.ClearRunningTask(); err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "clearing running task '%s' for pod '%s'", t.Id, p.ID))
+	}
+
 	if t.Requester == evergreen.MergeTestRequester {
 		if err = model.HandleEndTaskForCommitQueueTask(t, h.details.Status); err != nil {
-			return gimlet.MakeJSONErrorResponder(err)
+			return gimlet.MakeJSONInternalErrorResponder(err)
 		}
 	}
 
 	// the task was aborted if it is still in undispatched.
 	// the active state should be inactive.
 	if h.details.Status == evergreen.TaskUndispatched {
 		if t.Activated {
-			grip.Warningf("task %v is active and undispatched after being marked as finished", t.Id)
+			grip.Warningf("task %s is active and undispatched after being marked as finished", t.Id)
 			return gimlet.NewJSONResponse(&apimodels.EndTaskResponse{})
 		}
-		grip.Infof(fmt.Sprintf("task %v has been aborted and will not run", t.Id))
+		grip.Info(message.Fields{
+			"message": fmt.Sprintf("task %s has been aborted", t.Id),
```

Comment:
Usually for structured logging, we don't need to include the task ID in the message because the task ID is already included as one of the fields. The message is generally reserved for additional information about why it logged, whereas other fields like `task_id` are for specific resource identifiers that we want to include with the message.